### PR TITLE
Fortran Modules done

### PR DESCRIPTION
**Pull request by cyrillemidingoyi**
Some Crop modules shared by DSSAT on its'github repository are implemented  in  fortran 90. So, I write the modules in this version #1

### CODE REVIEW

File: src/f90/IsMomentRegistredZC_39.f90 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+    !use crop2mlModules
+
+        if (ANY(calendarMoments=="FlagLeafLiguleJustVisible"))then
```
Comment (pradal) [inline src/f90/IsMomentRegistredZC_39.f90]: Add a space before then

File: src/f90/VernalizationProgress.f90 [hunk 2]
```diff
@@ -0,0 +1,47 @@
+
+    !use crop2mlModules
+
+        real:: TT, maxVernaProg, DLverna, primordno, minLeafNumber, potlfno
+
+        ! Fortran Snippets
+        if ((IsVernalizable==1) .AND. (Vernaprog < 1)) then
+            TT = DeltaTT
+            if ((TT >= MinTvern) .AND. (TT <= IntTvern)) then
+
```
Comment (pradal) [inline src/f90/VernalizationProgress.f90]: Why an empty line?

File: src/f90/VernalizationProgress.f90 [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                Vernaprog = Vernaprog + VAI * TT + VBEE
+            end if
+
+            if (TT > IntTvern) then
+
```
Comment (pradal) [inline src/f90/VernalizationProgress.f90]: Remove empty line in the if statement

File: src/f90/crop2mlModules.f90 [hunk 4]
```diff
@@ -0,0 +1,111 @@
+module crop2mlModules
+
+contains
+    function indice(vectorElem, elem)  ! retrieve the index of an element in a vector
+	implicit none
+        character(len=*), dimension(:):: vectorElem
+        integer::iterator, indice
+        character(len=*):: elem
+        DO iterator= 1, size(vectorElem)
```
Comment (pradal) [inline src/f90/crop2mlModules.f90]: IF and ENDIF here are uppercase while before (other file) it was lower case.
Be consistent in your style.

Comment (pradal) [reply src/f90/crop2mlModules.f90]: The same for DO & al.

File: src/f90/leafNumber.f90 [hunk 5]
```diff
@@ -0,0 +1,45 @@
+
+    !use crop2mlModules
```
Comment (pradal) [inline src/f90/leafNumber.f90]: Is it a standard way to call external libs in Crop2ML or just a transient hack?

File: src/f90/updatePhase.f90 [hunk 6]
```diff
@@ -0,0 +1,156 @@
+
+    !use crop2mlModules
+        real:: phase1, appFLN, ttFromLastLeafToHeading, LocalDegfm, ttFromLastLeafToAnthesis
+
+        phase1 = phase;
+        if ((phase1 >= 0) .AND. (phase1 < 1))then
```
Comment (pradal) [inline src/f90/updatePhase.f90]: add space before then

File: src/f90/updatePhase.f90 [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+                end if
+            else
+                if (cumulTT >= Dse) then
+                    phase= 1;
+                else
+                    phase = phase1
+                end if
+            end if
+        else if ((phase1 >= 1) .AND. (phase1 < 2)) then
+            if ((IsVernalizable==1 .AND. Vernaprog >= 1) .OR. (IsVernalizable==0)) then
```
Comment (pradal) [inline src/f90/updatePhase.f90]: remove blank line

Comment (pradal) [reply src/f90/updatePhase.f90]: or add them in a consistent way

File: src/f90/updatePhase.f90 [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+                phase = phase1
+            end if
+
+
+        else if ((phase1 >= 5) .AND. (phase1 < 6)) then
+
+
+            LocalDegfm = Degfm
+            if (IgnoreGrainMaturation .eqv. .TRUE.) then
```
Comment (pradal) [inline src/f90/updatePhase.f90]: Some variables have the first char uppercase (IgnoreGrainMaturation) other lowercase (cumulTTFromZC).
Need consistency

File: test/f90/VernalizationProgress.f90 [hunk 9]
```diff
@@ -0,0 +1,67 @@
+module vernalizationProgressModule
+    use crop2mlModules
+    implicit none
+
+contains
+    subroutine calculateVernalizationProgress( DayLength, DeltaTT, cumulTT, &
```
Comment (pradal) [inline test/f90/VernalizationProgress.f90]: Here subroutine began with lowercase

File: test/f90/crop2mlModules.f90 [hunk 10]
```diff
@@ -0,0 +1,111 @@
+module crop2mlModules
```
Comment (pradal) [inline test/f90/crop2mlModules.f90]: Duplicated file from src...
Perhaps copy the files before testing but not two same file in git

**Review by pradal — CHANGES_REQUESTED** [review]
Mainly the first problem I saw is a problem of consistent coding style:
- use line break consistently (after a subroutine, after end if ot whatever else position
- Naming convention should also be consistent
- DO END DO or IF END IF also need consistency

File: src/f90/crop2mlModules.f90 (hunk 4 above)
Comment (cyrillemidingoyi) [reply src/f90/crop2mlModules.f90]: No difference in fortran but I can change it at the same

File: src/f90/leafNumber.f90 (hunk 5 above)
Comment (cyrillemidingoyi) [reply src/f90/leafNumber.f90]: yes, it is a standard way, but here I comment this line ("!") to show that some external libs are used in this module

File: src/f90/updatePhase.f90 (hunk 8 above)
Comment (cyrillemidingoyi) [reply src/f90/updatePhase.f90]: I didn't change the name of variables in the original Sirius Model

**Pull request merged**